mrbean/src/main/java/com/fasterxml/jackson/module/mrbean/BeanBuilder.java

```diff
@@ -176,16 +176,28 @@ protected boolean hasConcreteOverride(Method m0, JavaType implementedType)
     {
         final String name = m0.getName();
         final Class<?>[] argTypes = m0.getParameterTypes();
+        try {
```

Comment:
I guess I am specifically trying to understand this part here; why is the extra check needed?
Is that because existing code seems to only look for supertypes (and not type itself) or something else?

Reply:
It's a bit convoluted (and took me some time to spot the issue), so let me illustrate with the `BeanWithInheritedDefault` example from the unit test:

In [`BeanBuilder.implement`](https://github.com/FasterXML/jackson-modules-base/blob/73b8f0ca5b0ad2454eabbd7b6669603a70782a6b/mrbean/src/main/java/com/fasterxml/jackson/module/mrbean/BeanBuilder.java#L67),`ArrayList<JavaType> implTypes` contains `[BeanWithInheritedDefault, BeanWithDefaultForOtherInterface, Bean, OtherInterface]`, and these reduce two relevant executions of the loop with `Method m`:
  1. `BeanWithDefaultForOtherInterface.anyValuePresent`, which *is concrete* itself, and is properly skipped at [line 108](https://github.com/FasterXML/jackson-modules-base/blob/73b8f0ca5b0ad2454eabbd7b6669603a70782a6b/mrbean/src/main/java/com/fasterxml/jackson/module/mrbean/BeanBuilder.java#L108)
  2. `OtherInterface.anyValuePresent`, which is *not*, and therefore which calls the `hasConcreteOverride` method I've modified.

In the original implementation of `hasConcreteOverride`, it checks for a concrete declared method on the `implementedType` (in this case `BeanWithInheritedDefault`, which doesn't have it), and then it checks on super*classes* of which there is none.  Critically, this code fails to locate the concrete implementation of this method in `BeanWithDefaultForOtherInterface`, and therefore it incorrectly return false from the method.

I started to explore solutions that would either (a) traverse the interface hierarchy, or (b) optimize the loop in `implement` to track known-concrete methods, but I ultimately realized this simple addition to `hasConcreteOverride` solves the search problem for all public methods, which coincidentally applies to all Java interfaces and their default methods.  I think it also makes the `hasConcreteOverride` more efficient, since it is only necessary to search the superclass hierarchy when the method is in question is non-public.

Reply:
Ah ok. So the logic was ok pre-Java8, but the introduction of interface default implementations threw the monkey wrench... makes sense.